### ISSUE TITLE
linter: `typescript/no-unsafe-assignment` false positive on Vitest `expect.any()`

### ISSUE THREAD

**Issue by xeho91** [oxc-project/oxc#24668]
## What version of Oxlint are you using?

```
oxlint@1.74.0
oxlint-tsgolint@0.25.0
```

## What command did you run?

```
oxlint --type-aware
```

## What does your `oxlint.config.ts` config file look like?

```typescript
import { defineConfig } from "oxlint";

export default defineConfig({
  plugins: ["typescript", "vitest"],
  options: {
    typeAware: true,
  },
  rules: {
    "typescript/no-unsafe-assignment": "error",
  },
});
```

## What happened?

The [`typescript/no-unsafe-assignment` rule](https://oxc.rs/docs/guide/usage/linter/rules/typescript/no-unsafe-assignment.html) reports a false positive when using Vitest's `expect.any(Function)` asymmetric matcher.

Vitest's `expect.any()` intentionally returns `any` by design — it's an asymmetric matcher that matches any value of the given constructor type. The rule incorrectly flags this as an unsafe assignment.

## Reproduction

```typescript
import { describe, expect, it } from "vitest";

describe("example", () => {
  it("reproduces the issue", () => {
    const result = { /* ... */ }

    expect(result).toStrictEqual({
      filepath: "...",
      get_json: expect.any(Function), // ← false positive here
    });
  });
});
```

## Reported error

```
error typescript(no-unsafe-assignment): Unsafe assignment of an any value.
```

## Why this seems incorrect

The `expect.any()` matcher is a standard Vitest/Jest API that intentionally returns `any`.
This is not an actual unsafe assignment — it's an asymmetric matcher used in test assertions.

This is a known issue: [vitest-dev/vitest#6614](https://github.com/vitest-dev/vitest/discussions/6614)

## Suggested fix

The rule should recognize Vitest [`expect.any()`](https://vitest.dev/api/expect.html#expect-any) and a similar one [`expect.anything`](https://vitest.dev/api/expect.html#expect-anything).

**Comment by xeho91** [comment]
Sub-issue created: [oxc-project/tsgolint#1096](https://github.com/oxc-project/tsgolint/issues/1096)

**Comment by xeho91** [comment]
Closing, see: https://github.com/oxc-project/tsgolint/issues/1096
